syntax = "proto3";

package context_service;

import "google/protobuf/empty.proto";

// Context Resolution Service
service ContextService {
    // Resolve references in user message
    rpc ResolveReference(ReferenceRequest) returns (ReferenceResponse);
    
    // Store conversation turn for context tracking
    rpc StoreConversationTurn(ConversationTurnRequest) returns (ConversationTurnResponse);
    
    // Get proactive business data
    rpc GetProactiveData(ProactiveDataRequest) returns (ProactiveDataResponse);
    
    // Health check
    rpc Health(google.protobuf.Empty) returns (HealthResponse);
}

message ReferenceRequest {
    string message = 1;
    string session_id = 2;
    string tenant_id = 3;
}

message ReferenceResponse {
    string resolved_message = 1;
    repeated ResolvedReference references = 2;
    double confidence = 3;
    bool success = 4;
    string suggested_intent = 5;  // ‚Üê ADD THIS LINE
}

message ResolvedReference {
    string reference = 1;
    string entity_type = 2;
    int32 entity_id = 3;
    string entity_value = 4;
    double confidence = 5;
}

message ConversationTurnRequest {
    string session_id = 1;
    string tenant_id = 2;
    string user_message = 3;
    string system_response = 4;
    map<string, string> entities = 5;
    repeated ResolvedReference resolved_references = 6;
    string topic = 7;
    string action = 8;
}

message ConversationTurnResponse {
    bool success = 1;
    string message = 2;
}

message ProactiveDataRequest {
    repeated string topics = 1;
    string tenant_id = 2;
}

message ProactiveDataResponse {
    map<string, BusinessData> data = 1;
    bool success = 2;
}

message BusinessData {
    string data_type = 1;
    string current_value = 2;
    string source = 3;
    string last_updated = 4;
    repeated string search_terms = 5;
}

message HealthResponse {
    string status = 1;
}